### ISSUE TITLE
[flutter_markdown] Feature Request: Custom Widget for h1, h2 etc.

### ISSUE THREAD

**Issue by tomkuschka** [flutter/flutter#89153]
I just started to look into this package and wanted to implement a custom styling (Padding and underline that requires a stack) of h1 - h6. After briefly trying out to pass a custom MarkdownBuilder to the builder function of `MarkdownBody` like this `{'h1': _H1Builder(),` I realized that I'm not able to do this because h1 - h6 are defined in `_kBlockTags`. So I can't use a custom builder here.

Is there a way how I can achieve this behavior with the current implementation?

## Proposal

My proposal would be to check if the tag is inside the builder map. If yes use the provided builder, if not use the custom one for builder tags e.g.

**Comment by creativecreatorormaybenot** [comment]
@domesticmouse Any idea how this could be achieved?

**Comment by domesticmouse** [comment]
@creativecreatorormaybenot Nope. I'm just making sure that PRs have good test coverage and the changes pass the sniff test.

@tomkuschka If the problem is something is defined in an enumeration or similar I hereby give you the option of changing it to be a enumeration style class or something similar. Just make sure the tests pass, and your new code has test coverage. 😎

**Comment by creativecreatorormaybenot** [comment]
@domesticmouse Who is codeowner for flutter_markdown / who knows the implementation? Would be useful to get some insights.

**Comment by domesticmouse** [comment]
I'm the code owner in as much as I make sure the build stays sane. To figure out who to chat to about which part of the codebase you can go digging in the original repo - https://github.com/flutter/flutter_markdown

**Comment by creativecreatorormaybenot** [comment]
@domesticmouse No, I cannot. I am aware that it was moved and I tried checking the blame, but the repo is not accessible to me.
<img width="736" alt="Screen Shot 2021-08-31 at 13 32 46" src="https://user-images.githubusercontent.com/19204050/131511701-be755b80-25a4-4919-b2a4-ef2e44f8de34.png">
